### PR TITLE
5 upgrades2

### PR DESCRIPTION
**Pull request by Denomycor**

### CODE REVIEW

**Comment by Denomycor** [comment]
[Decouple assigned gun for upgrades](https://trello.com/c/WgpqEjx0/15-decouple-assigned-gun-for-upgrades)

**Review by PedroTFX — APPROVED** [review]
Test font & Deprecated Classes

File: SewerGodot/assets/upgrades/testProjectile/scripts/TestProjectile.cs [hunk 1]
```diff
@@ -34,19 +34,21 @@ public TestProjectile(Gun gun)
...
     }
 
-    protected override void PrepareScene(Player player){
+    protected override void InstallImpl(Player player){
+        base.InstallImpl(player);
         projectileTemplate = GD.Load<PackedScene>("res://assets/upgrades/testProjectile/scenes/TestProjectile.tscn");
     }
 
-    protected override void RemoveScene(Player player){
+    protected override void RemoveImpl(Player player){
```
Comment (vicentecsousa) [inline SewerGodot/assets/upgrades/testProjectile/scripts/TestProjectile.cs:50]: why do a lot of functions are called `[Word]Impl`. This type of name is not specific at all.

File: SewerGodot/assets/upgrades/testProjectile/scripts/TestProjectile.cs [hunk 2]
```diff
@@ -34,19 +34,21 @@ public TestProjectile(Gun gun)
         return -1;
     }
 
-    public override void InitConnections(){
+    protected override void InitConnections(){
         this.connectionsMap = new Dictionary<Godot.Vector2, int>();
         connectionsMap.Add(Vector2.Up, 1);
```
Comment (vicentecsousa) [inline SewerGodot/assets/upgrades/testProjectile/scripts/TestProjectile.cs:39]: what is a `connectionsMap` and why do we need it inside `TestProjectile` class

File: SewerGodot/assets/upgrades/testProjectile/scripts/TestProjectile.cs (hunk 1 above)
Comment (Denomycor) [reply SewerGodot/assets/upgrades/testProjectile/scripts/TestProjectile.cs:50]: Only two functions have the Impl suffix. RemoveImpl and Install impl, both are the IMPLementations of the functions Remove and Install, this providing a definition to the installation and Removal of pieces is mandatory, but the the standalone function automatically updates the state variable

File: SewerGodot/assets/upgrades/testProjectile/scripts/TestProjectile.cs (hunk 2 above)
Comment (Denomycor) [reply SewerGodot/assets/upgrades/testProjectile/scripts/TestProjectile.cs:39]: connectionsMap specifies the type of connections this upgrade has and their directions, on the Upgrade Menu

**Pull request merged**